### PR TITLE
[embedlite-components] Cleanup EmbedLiteConsoleListener. Fixes JB#55147 OMP#JOLLA-278

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

**Review by mkenttala — APPROVED** [review]

File: jscomps/EmbedLiteConsoleListener.js [hunk 1]
```diff
@@ -186,8 +166,8 @@ $EmbedLiteConsoleListener.prototype = {
         if (Logger.enabled) {
           this._listener = new SPConsoleListener();
           Services.console.registerListener(this._listener);
-          Services.obs.addObserver(this, "embedui:logger", true);
         }
+        Services.obs.addObserver(this, "embedui:logger", true);
```
Comment (chriadam) [inline jscomps/EmbedLiteConsoleListener.js:170]: I assume this was intentional.  Can you describe briefly what effect this now has?  I.e. in the Logger.enabled=false case.

Comment (rainemak) [reply jscomps/EmbedLiteConsoleListener.js:170]: Yes, intentional. We do not have front end for logging at the browser side (never have had such feature). That said, in order to have such a feature and being able to disable/enable logging from the front end, we need to be observing "embedui:logger" messages all the time regardless of the env variables.

File: jscomps/EmbedLiteConsoleListener.js [hunk 2]
```diff
@@ -202,14 +182,9 @@ $EmbedLiteConsoleListener.prototype = {
       case "embedui:logger": {
         var data = JSON.parse(aData);
         if (data.enabled) {
```
Comment (rainemak) [inline jscomps/EmbedLiteConsoleListener.js:184]: Related to the comment https://github.com/sailfishos/embedlite-components/pull/19#discussion_r699142353 if browser app sends that it wants to listen console log when we should respect that regardless of the env variables.

**Review by okodron — APPROVED** [review]

File: jscomps/EmbedLiteConsoleListener.js (hunk 1 above)
Comment (chriadam) [reply jscomps/EmbedLiteConsoleListener.js:170]: Would be good if such information could be added to the commit message at least

Comment (rainemak) [reply jscomps/EmbedLiteConsoleListener.js:170]: Added:
```
After https://github.com/sailfishos/embedlite-components/pull/8 there is
no need to cache messages. This commit cleans up caches and related functions.

We do not have frontend for logging at the browser side (never have had
such feature). That said, in order to have such a feature and being able to
disable/enable logging from the front end, we need to be observing "embedui:logger"
messages all the time regardless of the EMBED_CONSOLE variable values.
```

**Comment by rainemak** [comment]
Rebased

**Pull request merged**